test/integration-tests/IntegrationTests.StartupHook/StartupHookTests.cs

```diff
@@ -24,6 +24,30 @@ public StartupHookTests(ITestOutputHelper output)
         [Fact]
         [Trait("Category", "EndToEnd")]
         public void SubmitsTraces()
+        {
+            RunTestAndValidateResults(2);
+        }
+
+        [Theory]
+        [Trait("Category", "EndToEnd")]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ApplicationIsExcluded(bool excludeApplication)
+        {
+            string applicationNameToExclude = string.Empty;
+            int expectedSpanCount = 2;
+            if (excludeApplication)
+            {
+                applicationNameToExclude = EnvironmentHelper.FullSampleName;
+                expectedSpanCount = 0;
+            }
+
+            SetEnvironmentVariable("OTEL_PROFILER_EXCLUDE_PROCESSES", $"dotnet,dotnet.exe,{applicationNameToExclude}");
+
+            RunTestAndValidateResults(expectedSpanCount);
+        }
+
+        private void RunTestAndValidateResults(int expectedSpanCount)
```

Comment:
Proposal no 1: rename to `RunSampleAndAssertSpanCount`

Proposal no 2: refactor the tests so we have more readable tests like:

```csharp
using (var collector = RunSample(startupHook: true))
{
    AssertSpanCount(expectedSpanCount, collector);
}
```

then I think having a helper to extract two lines would be useless. Also, I find a method having `And` as a code smell.

Reply:
Yes, that part of the code definitely needs some cleaning up.

Reply:
I'll get this cleaned up in a follow up PR.